typescript-carousel/main.ts

```diff
@@ -0,0 +1,127 @@
+
+const $carouselContainer = document.querySelector(
+  '.carousel-container'
+) as HTMLDivElement;
+if (!$carouselContainer) throw new Error('.carousel-container query failed!');
+
+const $leftArrow = document.querySelector('.fa-chevron-left');
+if (!$leftArrow) throw new Error('.fa-chevron-left query failed!');
+const $rightArrow = document.querySelector('.fa-chevron-right');
+if (!$rightArrow) throw new Error('.fa-chevron-right query failed!');
+
+const $circles = document.querySelectorAll('.fa-circle');
+if (!$circles) throw new Error('.fa-circles query failed!');
+
+$carouselContainer.addEventListener('click', function (event: Event) {
+  if (event.target === $leftArrow) {
+    moveLeft();
+  }
+
+  if (event.target === $rightArrow) {
+    moveRight();
+  }
+
+  switch (event.target) {
+    case $circles[0]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(0);
+      break;
+    }
+    case $circles[1]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(1);
+      break;
+    }
+    case $circles[2]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(2);
+      break;
+    }
+    case $circles[3]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(3);
+      break;
+    }
+    case $circles[4]: {
+      clearInterval(intervalId);
+      intervalId = setInterval(intervalFunc, 3000);
+      clearCircles();
+      setNdxImageAndDot(4);
+      break;
+    }
+  }
+});
+
+let intervalId = setInterval(intervalFunc, 3000);
+
+function clearCircles(): void {
+  for (let i = 0; i < $circles.length; i++) {
+    $circles[i].classList.remove('fa-solid');
+    $circles[i].classList.add('fa-regular');
+  }
+}
+
+function setNdxImageAndDot(index: number): void {
+  currNdx = index;
+  $displayImage.setAttribute('src', images[index]);
+  $circles[index].classList.add('fa-solid');
+  $circles[index].classList.remove('fa-regular');
+}
+
+function intervalFunc(): void {
+  clearCircles();
+  $circles[currNdx].classList.add('fa-regular');
+  currNdx++;
+  if (currNdx === 5) {
+    currNdx = 0;
+  }
+  setNdxImageAndDot(currNdx);
+}
+
+function moveRight(): void {
+  clearInterval(intervalId);
+  intervalId = setInterval(intervalFunc, 3000);
+  if (currNdx <= 3) {
+    clearCircles();
+    setNdxImageAndDot(currNdx + 1);
+  } else if (currNdx === 4) {
+    clearCircles();
+    setNdxImageAndDot(0);
+  }
+}
+
+function moveLeft(): void {
+  clearInterval(intervalId);
+  intervalId = setInterval(intervalFunc, 3000);
+  if (currNdx >= 1) {
+    clearCircles();
+    setNdxImageAndDot(currNdx - 1);
+  } else if (currNdx === 0) {
+    clearCircles();
+    setNdxImageAndDot(4);
+  }
```

Comment:
You should update your index and show _before_ you restart the interval. What is something in your code after was necessary for your `intervalFunc`?